pkg/resource/properties.go

```diff
@@ -194,7 +194,7 @@ func MakeOutput(v PropertyValue) PropertyValue {
 }
 
 func MakeSecret(v PropertyValue) PropertyValue {
-	return NewSecretProperty(Secret{Element: v})
+	return NewSecretProperty(&Secret{Element: v})
```

Comment:
If I understand correctly, we need `NewSecretProperty` to take a pointer to a `resource.Secret` so we can use pointer equality to check if the secret is found in the `cachingSecretsManager`, right?

That approach is fine, but I wonder if we should instead -- or perhaps not now but later -- annotate the `resource.Secret` with some ID or tie it to a URN-like thing? The approach we are taking now would be difficult to replicate in other situations, for example if trying to use the Pulumi codebase as a library for some other client.

So I wonder if a better approach would be to provide each secret with some separate identity. Not worth blocking this PR on, just want to know what you think.

Reply:
> The approach we are taking now would be difficult to replicate in other situations, for example if trying to use the Pulumi codebase as a library for some other client.

Can you elaborate? I don't really understand what you're getting at here.

Reply:
This isn't a big concern or anything, more of a "is this a code smell?" type question.

By passing pointers around and using their addresses to check if we have cached the decrypted value or not, we are heavily relying on something very implicit here.

If the implementation of `NewSecretProperty` were to wrap the `resource.Secret` or `resource.PropertyValue` in some other type then it's possible our caching solution would stop working. That's probably unlikely and maybe not even possible.

But what I'm getting at is that rather than perhaps adding a comment "IMPORTANT! Beside to not mess with these pointers, because things won't get cached if you copy the values and change their addresses." it might be better to instead formalize what we need. Specifically, that every secret is given some stable identifiers.

Currently that is the address of `resource.Secret`, but you can imagine it being `resource.Secret::ID` or something else. That's perhaps a bigger, more tedious refactoring than what we need or want. But It seems like that might not be a bad idea, since then it's clear what we are depending on.

Does that make sense?

Reply:
> Does that make sense?

Yes, that makes sense. I'll add a comment.